pkg/apis/authorization/types.go

```diff
@@ -0,0 +1,120 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorization
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
+// spec.namespace means "in all namespaces".
+type SubjectAccessReview struct {
+	unversioned.TypeMeta
+
+	// Spec holds information about the request being evaluated
+	Spec SubjectAccessReviewSpec
+
+	// Status is filled in by the server and indicates whether the request is allowed or not
+	Status SubjectAccessReviewStatus
+}
+
+// SelfSubjectAccessReview checks whether or the current user can perform an action.  Not filling in a
+// spec.namespace means "in all namespaces".  Self is a special case, because users should always be able
+// to check whether they can perform an action
+type SelfSubjectAccessReview struct {
+	unversioned.TypeMeta
+
+	// Spec holds information about the request being evaluated.
+	Spec SelfSubjectAccessReviewSpec
+
+	// Status is filled in by the server and indicates whether the request is allowed or not
+	Status SubjectAccessReviewStatus
+}
+
+// LocalSubjectAccessReview checks whether or not a user or group can perform an action in a given namespace.
+// Having a namespace scoped resource makes it much easier to grant namespace scoped policy that includes permissions
+// checking.
+type LocalSubjectAccessReview struct {
+	unversioned.TypeMeta
+
+	// Spec holds information about the request being evaluated.  spec.namespace must be equal to the namespace
+	// you made the request against.  If empty, it is defaulted.
+	Spec SubjectAccessReviewSpec
+
+	// Status is filled in by the server and indicates whether the request is allowed or not
+	Status SubjectAccessReviewStatus
+}
+
+type ResourceAttributes struct {
+	// Namespace is the namespace of the action being requested.  Currently, there is no distinction between no namespace and all namespaces
+	// "" (empty) is defaulted for LocalSubjectAccessReviews
+	// "" (empty) is empty for cluster-scoped resources
+	// "" (empty) means "all" for namespace scoped resources from a SubjectAccessReview or SelfSubjectAccessReview
+	Namespace string
+	// Verb is a kubernetes resource API verb, like: get, list, watch, create, update, delete, proxy.  "*" means all.
+	Verb string
+	// Group is the API Group of the Resource.  "*" means all.
+	Group string
+	// Version is the API Version of the Resource.  "*" means all.
+	Version string
+	// Resource is one of the existing resource types.  "*" means all.
+	Resource string
+	// Subresource is one of the existing resource types
+	Subresource string
+	// Name is the name of the resource being requested for a "get" or deleted for a "delete". "" (empty) means all.
+	Name string
+}
+
+type NonResourceAttributes struct {
```

Comment:
Documentation for this type? Is it used? (I mean, I know it's in the struct below, but is it something we actually intend to implement?)